src/Inkluzitron/Services/RuntimeService.cs

```diff
@@ -31,6 +35,57 @@ public class RuntimeService
             Cache = cache;
 
             DiscordClient.Ready += OnReadyAsync;
+            DiscordClient.MessageUpdated += OnMessageUpdatedAsync;
+            DiscordClient.MessageDeleted += OnMessageDeletedAsync;
+            DiscordClient.MessagesBulkDeleted += OnMessagesBulkDeletedAsync;
+            DiscordClient.GuildAvailable += OnGuildAvailableAsync;
+        }
+        
+        private async Task OnGuildAvailableAsync(SocketGuild arg)
+        {
+            if (arg.Id.ToString() != Configuration["HomeGuildId"])
+                return;
+            else if (GuildReadyStateAnnounced)
+                return;
+
+            GuildReadyStateAnnounced = true;
+
+            foreach (var runtimeEventHandler in ServiceProvider.GetServices<IRuntimeEventHandler>())
+                await runtimeEventHandler.OnHomeGuildReadyAsync();
+        }
+
+        private async Task OnMessageUpdatedAsync(Cacheable<IMessage, ulong> oldMessage, SocketMessage newMessage, ISocketMessageChannel channel)
+        {
+            var freshMessageFactory = new Lazy<Task<IMessage>>(() => channel.GetMessageAsync(newMessage.Id));
+
+            foreach (var messageEventHandler in ServiceProvider.GetServices<IMessageEventHandler>())
+            {
+                var handled = await messageEventHandler.HandleMessageUpdatedAsync(channel, newMessage, freshMessageFactory);
+                if (handled)
+                    break;
+            }
+        }
+
+        private async Task OnMessageDeletedAsync(Cacheable<IMessage, ulong> deletedMessage, ISocketMessageChannel channel)
+        {
+            foreach (var messageEventHandler in ServiceProvider.GetServices<IMessageEventHandler>())
+            {
+                var handled = await messageEventHandler.HandleMessageDeletedAsync(channel, deletedMessage.Id);
+                if (handled)
+                    break;
+            }
+        }
+
+        private async Task OnMessagesBulkDeletedAsync(IReadOnlyCollection<Cacheable<IMessage, ulong>> deletedMessages, ISocketMessageChannel channel)
+        {
+            var messageIds = deletedMessages.Select(msg => msg.Id).ToList();
```

Comment:
Spíš bych předával IEnumerable. Kdyby se za tom dělo něco dalšího, tak ať to projede v "jednom cyklu"

Reply:
To bys ale musel předávat přímo IEnumerator, ne IEnumerable, a navíc bys tím zamezil jakémukoliv "bulk" zpracování, tzn. by to všichni handleři museli nutně zpracovávat pojednom. Tady jsem si mohl vybrat, že budu línej a nebudu to dělat hromadně.